### PR TITLE
Full online

### PR DESCRIPTION
**Pull request by gaetschwartz**
- Add a login page
- Fix functions + a lot more tests

### CODE REVIEW

File: app/src/main/java/ch/epfl/sweng/rps/LoadingActivity.kt [hunk 1]
```diff
@@ -0,0 +1,46 @@
+package ch.epfl.sweng.rps
```
Comment (panadax) [inline app/src/main/java/ch/epfl/sweng/rps/LoadingActivity.kt:1]: Could you explain more about this class, what is this used for?

Comment (gaetschwartz) [reply app/src/main/java/ch/epfl/sweng/rps/LoadingActivity.kt:1]: Basically, for different reasons (using firebase emulators etc) I had to disable Firebase automatic initializing. It thus had to be done somewhere, here. Also other logic can happen in the future (everything needing initialization)

Comment (panadax) [reply app/src/main/java/ch/epfl/sweng/rps/LoadingActivity.kt:1]: Thanks for your explaination! Take care and have a rest bro :)

File: app/src/main/java/ch/epfl/sweng/rps/services/MatchmakingService.kt [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+        if (games.isEmpty()) {
+            return null
+        }
+        if (games.size > 1) {
+            Log.e(
+                "MatchmakingService",
+                "More than one game for user ${repo.getCurrentUid()}: ${games.map { it.id }}"
+            )
+        }
+        return ServiceLocator.getInstance().getGameServiceForGame(games[0].id)
```
Comment (adam-narozniak) [inline app/src/main/java/ch/epfl/sweng/rps/services/MatchmakingService.kt:43]: Hi, could you elaborate on when this class would be used? And how do we know that the game index zero would the current one?

File: app/src/main/java/ch/epfl/sweng/rps/services/MatchmakingService.kt [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+    }
+
+    sealed class QueueStatus {
+        object Queued : QueueStatus()
+        class Accepted(val gameService: GameService) : QueueStatus()
+    }
```
Comment (adam-narozniak) [inline app/src/main/java/ch/epfl/sweng/rps/services/MatchmakingService.kt:49]: Could you elaborate on when this class would be used? What is the reason to make it sealed?

**Review by adam-narozniak — COMMENTED** [review]
Just dropped two questions.

File: app/src/main/java/ch/epfl/sweng/rps/services/MatchmakingService.kt (hunk 3 above)
Comment (gaetschwartz) [reply app/src/main/java/ch/epfl/sweng/rps/services/MatchmakingService.kt:49]: [Sealed classes﻿](https://kotlinlang.org/docs/sealed-classes.html) allow to use these as [Union](https://en.wikipedia.org/wiki/Union_type) classes.

File: app/src/main/java/ch/epfl/sweng/rps/services/MatchmakingService.kt (hunk 2 above)
Comment (gaetschwartz) [reply app/src/main/java/ch/epfl/sweng/rps/services/MatchmakingService.kt:43]: Theoretically there should be only one current game played at a time. If there are more, we return one of them, the first. Once this game will be done, you can play the next one.

File: app/src/main/java/ch/epfl/sweng/rps/services/MatchmakingService.kt (hunk 3 above)
Comment (gaetschwartz) [reply app/src/main/java/ch/epfl/sweng/rps/services/MatchmakingService.kt:49]: It's used to have a status on the current state of the queue. Read more about [flows](https://kotlinlang.org/docs/flow.html) if you need.

**Review by adam-narozniak — APPROVED** [review]
Thanks, I'll get into more details and ask then if still have questions.

**Pull request merged**